dialer/bandit_test.go

```diff
@@ -281,11 +326,112 @@ func Test_differentArm(t *testing.T) {
 	}
 }
 
+func TestUpdateBanditRewards(t *testing.T) {
+	var tests = []struct {
+		name   string
+		given  map[string]banditMetrics
+		assert func(t *testing.T, dir string, err error)
+	}{
+		{
+			name: "it should update rewards file",
+			given: map[string]banditMetrics{
+				"test-dialer": {
+					Reward: 1.0,
+					Count:  1,
+				},
+			},
+			assert: func(t *testing.T, dir string, err error) {
+				assert.NoError(t, err)
+				f, err := os.Open(filepath.Join(dir, "rewards.csv"))
+				require.NoError(t, err)
+				defer f.Close()
+				b, err := io.ReadAll(f)
+				require.NoError(t, err)
+
+				lines := strings.Split(string(b), "\n")
+				// check if headers are there
+				assert.Contains(t, lines[0], "dialer,reward,count,updated at")
+				// check if the data is there
+				assert.Contains(t, lines[1], "test-dialer,1.000000,1")
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tempDir, err := os.MkdirTemp("", "client_test")
+			require.NoError(t, err)
+			defer os.RemoveAll(tempDir)
+
+			banditDialer := &BanditDialer{
+				opts: &Options{
+					BanditDir: tempDir,
+				},
+				banditRewardsMutex: new(sync.Mutex),
+			}
+			err = banditDialer.UpdateBanditRewards(tt.given)
+			tt.assert(t, tempDir, err)
+		})
+	}
+}
+
+func TestLoadLastBanditRewards(t *testing.T) {
+	now := time.Now().UTC().Unix()
+	var tests = []struct {
+		name   string
+		given  string
+		assert func(t *testing.T, metrics map[string]banditMetrics, err error)
+	}{
+		{
+			name:  "it should load the rewards",
+			given: fmt.Sprintf("dialer,reward,count,updated at\ntest-dialer,1.000000,1,%d\n", now),
+			assert: func(t *testing.T, metrics map[string]banditMetrics, err error) {
+				assert.NoError(t, err)
+				assert.Contains(t, metrics, "test-dialer")
+				assert.Equal(t, 1.0, metrics["test-dialer"].Reward)
+				assert.Equal(t, 1, metrics["test-dialer"].Count)
+				assert.Equal(t, now, metrics["test-dialer"].UpdatedAt)
+			},
+		},
+		{
+			name:  "it should ignore dialers with updated at greater than 7 days",
+			given: fmt.Sprintf("dialer,reward,count,updated at\ntest-dialer,1.000000,1,%d\n", now-60*60*24*8),
+			assert: func(t *testing.T, metrics map[string]banditMetrics, err error) {
+				assert.NoError(t, err)
+				assert.Empty(t, metrics)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tempDir, err := os.MkdirTemp("", "client_test")
+			require.NoError(t, err)
+			defer os.RemoveAll(tempDir)
+			require.NoError(t, os.MkdirAll(filepath.Join(tempDir, "bandit"), 0755), "unable to create bandit directory")
+
+			f, err := os.Create(filepath.Join(tempDir, "rewards.csv"))
+			require.NoError(t, err)
+			defer f.Close()
+			_, err = f.WriteString(tt.given)
```

Comment:
This could all be condensed to 
`err = os.WriteFile(filepath.Join(tempDir, "rewards.csv"), []byte(tt.given), 0644)`. :smiley: